pkg/tecdsa/dkg/member.go

```diff
@@ -256,9 +259,382 @@ func (fm *finalizingMember) MarkInactiveMembers(
 // Result can be either the successful computation of the distributed key
 // generation process or a notification of failure.
 func (fm *finalizingMember) Result() *Result {
+	misbehavedMembersAsBytes := func(
+		inactive []group.MemberIndex,
+		disqualified []group.MemberIndex,
+	) []byte {
+		// merge IA and DQ into 'misbehaved' set
+		misbehaving := make(map[group.MemberIndex]bool)
+		for _, ia := range inactive {
+			misbehaving[ia] = true
+		}
+		for _, dq := range disqualified {
+			misbehaving[dq] = true
+		}
+
+		// convert 'misbehaved' set into sorted list
+		var sorted []group.MemberIndex
+		for m := range misbehaving {
+			sorted = append(sorted, m)
+		}
+		sort.Slice(sorted[:], func(i, j int) bool {
+			return sorted[i] < sorted[j]
+		})
+
+		// convert sorted list of member indexes into bytes
+		bytes := make([]byte, len(sorted))
+		for i, m := range sorted {
+			bytes[i] = byte(m)
+		}
+
+		return bytes
+	}
+
+	privateKeyShare := tecdsa.NewPrivateKeyShare(fm.tssResult)
+	publicKey := privateKeyShare.PublicKey()
+
 	return &Result{
-		PrivateKeyShare: tecdsa.NewPrivateKeyShare(fm.tssResult),
+		Group: fm.group,
+		Misbehaved: misbehavedMembersAsBytes(
+			fm.group.InactiveMemberIDs(),
+			fm.group.DisqualifiedMemberIDs(),
+		),
+		// TODO: Check how the public key should be obtained
+		GroupPublicKeyBytes: elliptic.Marshal(
+			publicKey.Curve,
+			publicKey.X,
+			publicKey.Y,
+		),
+		PrivateKeyShare: privateKeyShare,
+	}
+}
+
+// signingMember represents a group member sharing their preferred DKG result hash
+// and signature (over this hash) with other peer members.
+type signingMember struct {
+	logger log.StandardLogger
+	index  group.MemberIndex
+	// Group to which this member belongs.
+	group *group.Group
+	// Validator allowing to check public key and member index
+	// against group members
+	membershipValidator *group.MembershipValidator
+	// DKG result submission configuration
+	config *SubmissionConfig
+	// Hash of DKG result preferred by the current participant.
+	preferredDKGResultHash ResultHash
+	// Signature over preferredDKGResultHash calculated by the member.
+	selfDKGResultSignature []byte
+}
+
+// SignDKGResult calculates hash of DKG result and member's signature over this
+// hash. It packs the hash and signature into a broadcast message.
+func (sm *signingMember) SignDKGResult(
```

Comment:
Let's move that to `protocol.go` and cover with unit tests.

Reply:
Done in 78fdd1ad719aa298a5bc9f10705288e5989d7571.